packages/server/src/unstable-core-do-not-import/stream/sse.ts

```diff
@@ -235,94 +277,125 @@ export function sseStreamConsumer<TConfig extends ConsumerConfig>(
+        eventSource.addEventListener(SERIALIZED_ERROR_EVENT, (_msg) => {
+          const msg = _msg as EventSourceLike.MessageEvent;
 
-        controller.enqueue({
-          type: 'serialized-error',
-          error: deserialize(JSON.parse(msg.data)),
-          eventSource,
+          controller.enqueue({
+            type: 'serialized-error',
+            error: deserialize(JSON.parse(msg.data)),
+            eventSource,
+          });
         });
-      });
-      eventSource.addEventListener('error', (event) => {
-        if (eventSource.readyState === EventSource.CLOSED) {
-          controller.error(event);
-        } else {
+        eventSource.addEventListener(PING_EVENT, () => {
           controller.enqueue({
-            type: 'connecting',
+            type: 'ping',
             eventSource,
-            event,
           });
-        }
-      });
-      eventSource.addEventListener('message', (_msg) => {
-        const msg = _msg as EventSourceLike.MessageEvent;
+        });
+        eventSource.addEventListener('error', (event) => {
+          if (eventSource.readyState === EventSource.CLOSED) {
+            controller.error(event);
+          } else {
+            controller.enqueue({
+              type: 'connecting',
+              eventSource,
+              event,
+            });
+          }
+        });
+        eventSource.addEventListener('message', (_msg) => {
+          const msg = _msg as EventSourceLike.MessageEvent;
 
-        const chunk = deserialize(JSON.parse(msg.data));
+          const chunk = deserialize(JSON.parse(msg.data));
 
-        const def: SSEvent = {
-          data: chunk,
-        };
-        if (msg.lastEventId) {
-          def.id = msg.lastEventId;
-        }
-        controller.enqueue({
-          type: 'data',
-          data: def as inferTrackedOutput<TConfig['data']>,
-          eventSource,
+          const def: SSEvent = {
+            data: chunk,
+          };
+          if (msg.lastEventId) {
+            def.id = msg.lastEventId;
+          }
+          controller.enqueue({
+            type: 'data',
+            data: def as inferTrackedOutput<TConfig['data']>,
+            eventSource,
+          });
         });
-      });
 
-      const onAbort = () => {
-        controller.close();
-        eventSource.close();
-      };
-      if (signal.aborted) {
-        onAbort();
-      } else {
-        signal.addEventListener('abort', onAbort);
-      }
-    },
-    cancel() {
-      _es?.close();
-    },
-  });
+        const onAbort = () => {
+          controller.close();
+          eventSource.close();
+        };
+        if (signal.aborted) {
+          onAbort();
+        } else {
+          signal.addEventListener('abort', onAbort);
+        }
+      },
+      cancel() {
+        _es?.close();
+      },
+    });
+  const stream = getStream();
   return {
     [Symbol.asyncIterator]() {
-      const reader = stream.getReader();
-
+      let reader = stream.getReader();
```

Comment:
i feel like i'm doing something wrong here -

i first created the reader as `let reader = getStream().getReader()` but it broke some tests

Reply:
if you wanna play you can run

```
p test-watch sse httpSub
```
to test all the relevant tests in watch mode